scripts/s3/s3api/test_multipart_io_stability.py

```diff
@@ -0,0 +1,155 @@
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+#
+"""File contains s3 multipart test script for io stability."""
+
+from __future__ import division
+import logging
+import os
+import random
+import hashlib
+from datetime import datetime, timedelta
+from time import perf_counter_ns
+from typing import Union
+from src.libs.s3api.s3_multipart_ops import S3MultiParts
+from src.libs.s3api.s3_object_ops import S3Object
+from src.libs.s3api.s3_bucket_ops import S3Bucket
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TestMultiParts(S3MultiParts, S3Object, S3Bucket):
+    """Multipart class for executing given io stability workload."""
+
+    # pylint: disable=too-many-arguments, too-many-locals, too-many-instance-attributes
+    # pylint: disable=too-few-public-methods, too-many-statements
+
+    def __init__(self, access_key: str, secret_key: str, endpoint_url: str, use_ssl: bool,
+                 object_size: Union[dict, bytes], part_range: dict, seed: int, test_id: str = None,
+                 range_read: Union[dict, bytes] = None, part_copy: bool = False,
+                 duration: timedelta = None) -> None:
+        """s3 multipart init for multipart, part copy operations with different workload.
+
+        :param access_key: access key.
+        :param secret_key: secret key.
+        :param endpoint_url: endpoint with http or https.
+        :param test_id: Test ID string.
+        :param use_ssl: To use secure connection.
+        :param object_size: Size of the object in bytes.
+        :param part_range: Number of parts to be uploaded from given range.
+        :param part_copy: Perform part copy if True else normal part upload.
+        :param duration: Duration timedelta object, if not given will run for 100 days.
+        """
+        super().__init__(access_key, secret_key, endpoint_url=endpoint_url, use_ssl=use_ssl)
+        random.seed(seed)
+        self.duration = duration
+        self.object_size = object_size
+        self.part_range = part_range
+        self.part_copy = part_copy
+        self.range_read = range_read
+        self.iteration = 1
+        self.min_duration = 10  # In seconds
+        self.test_id = test_id.lower() if test_id else str(random.randrange(24, 240))
+        self.finish_time = datetime.now() + duration if duration else datetime.now() + \
+            timedelta(hours=int(100 * 24))
+
+    async def execute_multipart_workload(self):
+        """Execute multipart workload for specific duration."""
+        mpart_bucket = "s3mpart-bkt-{}-{}".format(self.test_id, perf_counter_ns())
+        response = await self.create_bucket(mpart_bucket)
+        assert response["ResponseMetadata"]["HTTPStatusCode"] == 200, \
```

Comment:
Done assert with 200 status code. The code will never get executed.

Reply:
removed